enhancements/insights/config-api.md

```diff
@@ -0,0 +1,286 @@
+
+### Goals
+
+The goal of this enhancement is to propose a basic structure for a new Insights configuration API and define new configuration options. This includes following:
+
+- Define a new configuration option providing a way to trigger on demand Insights gathering
+- Define a new configuration option providing a way to include/exclude particular Insights data gatherers
+- Provide Insights status & reporting in the status field in the new API
+
+### Non-Goals
+
+- Move the majority of the current configuration options to a new configuration API - Insights has quite a lot of configuration options
+  and it does not make sense to expose everything in the new configuration API 
+- Removing the original configuration via `support` secret
+
+## Proposal
+
+The proposal is to introduce a new Insights `CustomResourceDefinition` in the OpenShift API providing a way to configure the Insights operator and to report its status. The new CRD should follow a common CRD structure providing `spec` and `status` attributes. 
+
+```go
+type InsightsSpec struct {
+	// GatheringConfig spec attribute includes all the configuration options related to
+	// gathering of the Insights archive and its uploading to the ingress.
+	GatheringConfig `json:"gatheringConfig,omitempty"`
+}
+
+type InsightsStatus struct {
+	// GatheringStatus provides basic information about the last Insights gathering.
+	GatheringStatus *GatheringStatus `json:"gatheringStatus,omitempty"`
+	// ReportStatus provides general Insights analysis results.
+	ReportStatus *ReportStatus `json:"reportStatus,omitempty"`
+}
+
+type GatheringConfig struct {
+	// EnableObfuscation is flag enabling the global obfuscation of the IP addresses and base domain
+	// in the Insights archive data.
+	// +kubebuilder:default=false
+	EnableObfuscation bool `json:"enableObfuscation,omitempty"`
+	// Enables user to force Insights data gathering by setting a new reason.
+	// When there is some gathering in the progress then it is interrupted.
+	// When all the gatherers are deactivated by the `DisabledGatherers`, nothing happens.
+	ForceGatherReason string `json:"forceGatherReason,omitempty"`
+	// List of gatherers to be excluded from the gathering. All the gatherers can disabled by providing "all" value.
+	// If all the gatherers are disabled, the Insights operator does not gather any data.
+	// The particular gatherers IDs can be found at https://github.com/openshift/insights-operator/blob/master/docs/gathered-data.md.
+	// An example of disabling some gatherers can look like `disabledGatherers: ["clusterconfig/machine_configs", "workloads/workload_info"]`
+	DisabledGatherers []string `json:"disabledGatherers,omitempty"`
```

Comment:
Maybe it's worth also including the option EnabledGatherers

Reply:
I think I would not complicate it with such option. We would have to clarify what takes precedence when a gatherer is enabled and disabled at the same time or when all is disabled and something is enabled. My feeling is that it would complicate things even more.....